### ISSUE TITLE
"extraKey" property on React component erases CodeMirror default key map behavior

### ISSUE THREAD

**Issue by jonrhall** [RIP21/react-simplemde-editor#120]
# Current Behavior
When a user defines the `extraKey` property on the `<ReactSimpleMDE />`  component, it overwrites/erases the default key handler behavior for the entire component such that the editor stops reacting to events like `Cmd-P` (Preview) and `F11` (Full screen editor), among others.

## Questions
1. Is this intended behavior? It seems from the CodeMirror documentation the `extraKeys` property is meant to add keymaps _in addition to_ the default keymap. [See documentation here](https://codemirror.net/doc/manual.html#option_extraKeys):

   > Can be used to specify extra key bindings for the editor, alongside the ones defined by keyMap.
2. _(Assuming the answer to #1 is "no")_ Is this behavior that react-simplemde-editor is causing, or is it a downstream dependency (i.e. - CodeMirror or SimpleMDE)?

## Workaround
Set the SimpleMDE `getInstance` to capture the ref, then add your own keymap on top of what the default codemirror behavior is:

```javascript
// Handler
getInstance = (simplemde) => {
  simplemde.codemirror.addKeyMap({
    Up: (cm) => {
      cm.replaceSelection(" surprise. ");
    },
    Down: (cm) => {
      cm.replaceSelection(" surprise again! ");
    }
  });
}

// JSX
<SimpleMDE
  getMdeInstance={getInstance}
  {...otherProps}
/>
```

## CodeMirror Repro

Here is the problem in action ([also viewable here](https://react-simplemde-edtior.netlify.com/)): https://codesandbox.io/s/compassionate-fire-db63m

And here is the problem fixed by my workaround: https://codesandbox.io/s/dark-mountain-51kuy

[Here is a screenshot of the problematic behavior in the online sample referenced in the README](https://i.imgur.com/nOQBfnD.png)

**Comment by mikejackowski** [comment]
Any news on this?

**Comment by RIP21** [comment]
I really need to spent few hours one day addressing those... Sorry, I'm building a startup as a CTO and Co-founder (basically doing all the shit in the world) now and I barely have time for myself let alone some library that I don't use for last 3 years :) So, I cannot promise, but I'll try do smth today (will be fun to read this 3 years after while this issue will be still open :D)
But I'll try folks, I'll try to.
Maybe it's time to rewrite it using hooks? Will see.

**Comment by ChesterZengJian** [comment]
Well, I had the same problem too. I will try to see what is causing this problem.

**Comment by ChesterZengJian** [comment]
Actually we don't need to  use the `extraKeys` function, We just set the shortcut with the options property as follows and this will eliminate the above problem. [More information can see the easyMde](https://github.com/Ionaru/easy-markdown-editor#keyboard-shortcuts)  
@jonrhall @RIP21 @mikejackowski   
![image](https://user-images.githubusercontent.com/22461405/115285888-ff893500-a180-11eb-8e67-4a414bd44d9f.png)

**Comment by RIP21** [comment]
@ChesterZengJian thanks for the workaround. Yeah, AFAIR extraKeys are spread to options. Maybe I shall introduce breaking change later on for such API if it's there as it's pretty useless then :)

**Comment by ChesterZengJian** [comment]
Thx

**Comment by ChesterZengJian** [comment]
I think I found out why our extraKeys doesn't work. Because EasyMde shortcuts are overwritten by us.  

@RIP21 

![image](https://user-images.githubusercontent.com/22461405/115558318-34ae9800-a2e5-11eb-9d1b-ac0154023b68.png)
![image](https://user-images.githubusercontent.com/22461405/115558403-4a23c200-a2e5-11eb-9582-1dcfb07312e4.png)

**Comment by RIP21** [comment]
fixed in #153 
I'll have to publish still tho.

**Comment by RIP21** [comment]
`4.1.4` has this fixed. Just published.
